client/package.json

```diff
@@ -26,9 +22,9 @@
     "react-spinners": "^0.9.0"
   },
   "scripts": {
-    "dev": "react-scripts start",
-    "start": "node server.js",
-    "build": "NODE_PATH=src react-scripts build",
+    "start": "react-scripts start",
+    "build": "react-scripts build",
+    "postbuild": "cp build/index.html build/404.html",
```

Comment:
Why is this step necessary? (I didn't find a 404.html file used in our original codebase)

Reply:
GitHub Pages will look for a matching file when a user requests a specific path, but Since React builds a single-page app, there are no files associated with these paths and GitHub Pages will then attempt to serve the `404.html`. We know that React Router is able to function from the `index.html` and associated bundled JavaScript, so we can can serve the same code when GitHub Pages thinks that the path does not exist.

This however does introduce a semantics issue since the HTTP status code will still be 404 which may cause some issues with SEO indexing.

You can read more about this in the [Notes on client-side routing](https://github.com/rafgraph/spa-github-pages) section of the Create React App Deployment documentation.

Reply:
Added a temporary workaround of symlinking existing paths to `index.html` in 78dca6c